### DISCUSSION TITLE
Time-based one-time password (TOTP) for email/password auth

### DISCUSSION THREAD

**Discussion by oldwired** [Budibase/budibase#5896] [Ideas]
Adding optional [TOTP](https://en.wikipedia.org/wiki/Time-based_one-time_password) (RFC 6238 - like implemented in Google and Microsoft Authenticator for example) to the email/password auth (with the option for admins to enforce it globally or per App) would be great. For some use cases the additional security would be great, but using "cloud authentication" like Google Account or OpenID would be not a good fit.

**Comment by melohagan** [comment]
Hey @oldwired 

Might be a stickler point, but I think TOTP is to do with authentication, so I don't think it's a good idea to use it on a per app basis because that's authorisation. I could be wrong though.

Do you have an example circumstance were OpenID isn't suitable?

**Reply by oldwired** [reply]
You are probably right regarding the "per App" part.

Having it built into the basic email/password auth option would just be easiest. Having to run (and learn about) an OpenID-solution would be additional overhead. I wouldn't even know where to start. Perhaps with this list: https://openid.net/developers/certified/#OPServices

**Reply by melohagan** [reply]
For Q3 of the roadmap, the plans to overhaul and expand on the RBAC system, so maybe TOTP for Budibase log in could be part of that.
